File: 283.移动零.cs
/*
 * @lc app=leetcode.cn id=283 lang=csharp
 *
 * [283] 移动零
 */

// @lc code=start
public class Solution {
    public void MoveZeroes(int[] nums) {
        if(nums.Length == 0) return;
        int p = 0, q = 0; //p - 顺序遍历指针; q - 非0值位置的指针
        while(p < nums.Length)
        {
            if(nums[p] == 0)
            {
                p++;
            }
            else
            {
                nums[q++] = nums[p++];
            }
        }
        while(q < nums.Length)
        {
            nums[q++] = 0;
        }
        return;
    }
}
// @lc code=end

File: Answer/283.移动零.cpp
/*
 * @lc app=leetcode.cn id=283 lang=cpp
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if(nums.size() < 2) return;
        int i = 0, j = 0;
        while(j < nums.size())
        {
            if(i != j) nums[i] = nums[j];
            if(nums[i] == 0)
            {
                j++;
                continue;
            }
            i++;
            j++;
        }
        for(int t = i; t < nums.size(); t++)
        {
            nums[t] = 0;
        }
        return;
    }
};
// @lc code=end

